Hmwk_#3_Hangman/hangman_game.py

```diff
@@ -0,0 +1,159 @@
+import functools
+import random
+from functools import reduce
+import time
+import hangman_graphics
+
+#get data from file but not read so later it can be accesed twice without reopening
+def get_words(filename):
+    text = open(filename, "r", encoding="utf-8")
+
+    return text
+
+#define mode PC vs. PC, Player vs. PC
+def set_game_mode():
+    game_mode = input("What type of game you wanna play? (Manual / Auto) ").lower()
+    
+    if game_mode == "manual":
+        print("Manual game mode requires to interact with PC !")
+    elif game_mode == "auto":
+        print("Auto game mode allows just to watch the flow")
+
+    return game_mode
+
+#counter for PC tries to guess word
+def guess_counter(guesses):
+
+    if guesses == 9:
+        print("You have last try left...!")
+    elif guesses == 10:
+        print("Out of tries, you've lost!")
+
+    return guesses
+
+#get word to guess from file randomly or define it manually
+def choose_word(text):
+
+    words = text
+    randomize = input("Do you want to choose word by yourself or select a word randmoly? (Choose / Random)...").lower()
+    wordlist = [word.lower().strip() for word in words.readlines()]
+
+    if randomize == 'choose':
+        word_to_guess =  input("Choose the word from to be guessed from a list")
+    elif randomize == 'random':
+        word_to_guess = random.choice(wordlist)
+
+    return word_to_guess
+
+#function to be used in 'reduce' to count characters frequency on fly
+def counter(letter_freq, char):
+
+    if char.isalpha() == True:
+        letter_freq[char] = letter_freq.get(char, 0) + 1 #count each appearance of character and sore it as a value for 'character' key in dict
+        
+    return letter_freq
+
+#wrapper function to use 'counter' for letters frequncy check
+def get_letter_freq(text):
+
+    text.seek(0) #go to beginning if the file to avoid reopening
+    characters = text.read()
+
+    return reduce (counter, characters, {}) #use 'counter'
+
+#finction to receive letter to try to guess based on frequency, letters stored as sorted ascending dictionary
+def pick_up_letter(letters_freq):
+
+    letter = ''
+    most_frequent = {k: v for k, v in sorted(letters_freq.items(), key=lambda item: item[1])} #sort ascending
+    for char, freq in most_frequent.items():
+        letter = char
+
+    return letter
+
+#chech if received letter is present in word_to_guess
+def letter_appearance(game_mode, word_to_guess, letter):
+
+    if game_mode == 'manual':
+        letter_approval = input("Is there a letter '%s' in a word? (Y / N) " % (letter))
+    elif game_mode == 'auto':
+        word_letters = set([ch for ch in word_to_guess])
+        print("PC: Is there a letter '%s' in a word? (Y / N) " % (letter))
+        if letter in word_letters:
+            letter_approval ='Y'    
+            print("Player: ", "'", letter_approval , "'"," Correct, you've guessed the letter!", '\n')
+        elif letter not in word_letters:
```

Comment:
Statements `elif letter not in word_letters:`  and `if letter in word_letters:` are interchangeable. 

It'd be better simply to write:

`else: `

instead of doing an unnecessary calculations.